Src/BootCamp.Chapter1/Hangman.cs

```diff
@@ -23,18 +24,76 @@ public static class Hangman
         /// <returns>True if game was won or false if game was lost.</returns>
         public static bool Play(int lives, string wordsFile, int difficulty)
         {
-            var wordToGuess = WordsBank.PickRandomWord(@$"Words/{wordsFile}", difficulty);
-            // ToDo: finish the game!
-            var playersGuess = GuessCharacter();
+            if(lives < 1 || difficulty < 3)
+            {
+                throw new SystemException();
+            }
 
-            return false;
+            var wordToGuess = WordsBank.PickRandomWord(@$"Words/{wordsFile}", difficulty);           
+            bool[] isVisible = new bool[wordToGuess.Length];
+
+            while (true)
+            {
+                var playersGuess = GuessCharacter(lives);
+
+                bool isCharacterPresent = CheckForCharacter(wordToGuess, playersGuess, ref isVisible);
+
+                PrintKnownCharacters(wordToGuess, isVisible);
+
+                if (!isCharacterPresent)
+                {
+                    lives--;
+                }
+
+                if (lives == 0)
+                {
+                    Console.WriteLine("Too bad! You lost.");
+                    Console.WriteLine($"The correct word was {wordToGuess}.");
+                    return false;
+                }
+
+                else if (isVisible.All(x => x == true))
```

Comment:
Invert condition and use Any(). All will check all entries to be true, Any will check if any is false. Any is more optimal.

Reply:
More optimal, because a single missmatch in condition will stop it.